addons/mail/static/tests/systray/systray_messaging_menu_tests.js

```diff
@@ -839,5 +839,44 @@ QUnit.test('messaging menu widget: do not open chat window twice on preview clic
     messagingMenu.destroy();
 });
 
+QUnit.test('messaging menu widget: expand on thread preview', function (assert) {
+    assert.expect(4);
+
+    var messagingMenu = new MessagingMenu();
+    var taskMessage = [{
+        author_id: [1, "Demo"],
+        body: "<p>Office Design Project</p>",
+        res_id: 123,
+        model: 'some.res.model',
+        needaction: true,
+        needaction_partner_ids: [44],
+    }];
+    this.data['mail.message'].records =
+        this.data['mail.message'].records.concat(taskMessage);
+
+    testUtils.mock.addMockEnvironment(messagingMenu, {
+        services: this.services,
+        data: this.data,
+        session: {
+            partner_id: 44,
+        },
+        intercepts: {
+            do_action: function (ev) {
+                assert.step('do_action');
+            },
+        },
+    });
+    messagingMenu.appendTo($('#qunit-fixture'));
+
+    testUtils.dom.click(messagingMenu.$('.dropdown-toggle'));
+
+    var $preview = messagingMenu.$('.o_mail_preview');
+    assert.containsOnce(messagingMenu, '.o_thread_window_expand', "should display one preview");
+    assert.strictEqual($preview.data('document-model'), "some.res.model", "preview should be from crm.lead");
+    $preview.find('.o_thread_window_expand').click();
+    assert.verifySteps(['do_action'],"the document thread should be marked as read");
```

Comment:
`do_action` opens the document, it doesn't mean it will be marked as read (this is a side-effect btw)

Reply:
Changes done.